enginetest/memory_engine_test.go

```diff
@@ -301,6 +301,9 @@ func TestAnsiQuotesSqlMode(t *testing.T) {
 }
 
 func TestAnsiQuotesSqlModePrepared(t *testing.T) {
+	if _, ok := os.LookupEnv("SERVER_ENGINE_TEST"); ok {
+		t.Skip("prepared test depend on context for current sql_mode information, but it does not get updated when using ServerEngine")
+	}
```

Comment:
This test and `TestDateParse` are skipped using the environment variable `SERVER_ENGINE_TEST` and it works on CI, but when you run the `go test`, these tests are not skipped like other tests because I cannot check using `IsServerEngine(engine)` as engines used for these tests are not available. I can expand these two tests to have engine defined in `enginetest.go` rather than single function call as of right now. What do you think?

Reply:
How about if you instantiated the `NewDefaultMemoryHarness` at the very top of this function and then you checked to see if it had the server option enabled and did the `t.Skip` based on that? Or you could grab the 

It's still not the exact same check as the other functions are using, but seems like it would make it work for both CI env and local dev envs.